### ISSUE TITLE
Different results for numpy.fft.fft() when passing different types of argument (DataFrame, np.array) with the same values

### ISSUE THREAD

**Issue by ThemisKoutsellis** [numpy/numpy#23874]
### Describe the issue:

From the attached CSV file, [input_data.csv](https://github.com/scipy/scipy/files/11653905/input_data.csv), I import the, say, `test_list`. Then, I create a `pd.Dataframe` to store my data in `test_df`.

```
import numpy as np

test_df = pd.DataFrame(index=range(len(test_list)), columns=['test'])
test_df['test'] = test_list
```
I want to calculate the FFT transform of the signal in `test_df['test']`. However, I get different results when I pass as input different types of arguments to the `np.fft.fft()`.

Once I pass the signal as a nympy.array I get the plot #1.

![dummy](https://github.com/scipy/scipy/assets/88409436/7cab8d6a-45e1-4e7f-a377-2d601b84c40b)

Once I pass the signal as a pd.Dataframe I get the plot #2.

![dummy](https://github.com/scipy/scipy/assets/88409436/41fb7740-5c23-4d59-b0f4-a7c5f20ac42c)

Both inputs, np.array and pd.DataFrame contain the same information.

From the problem definition the plot#2 seems to have a better physical interpretation.
However, it is not clear why I get different results.

May be it is a bug?

The code I use is below:

```
import numpy as np

test_df = pd.DataFrame(index=range(len(test_list)), columns=['test'])
test_df['test'] = test_list


########## PLOT 1: from Numpy  ##############

to_np = test_df['test'].to_numpy()
print(to_np)

fft_output = np.fft.fft(to_np)
power = np.abs(fft_output)
freq = np.fft.fftfreq(len(to_np))

mask = freq >= 0
freq = freq[mask]
power = power[mask]


plt.plot(freq, power)
plt.show()


########## PLOT 2: from pd.DataFrame  ##############

to_df = test_df['test'].to_frame()
print(to_df)

fft_output = np.fft.fft(to_df)
power = np.abs(fft_output)
freq = np.fft.fftfreq(len(to_df))


mask = freq >= 0
freq = freq[mask]
power = power[mask]


plt.plot(freq, power)
plt.show()

```

### Reproduce the code example:

```python
import numpy as np

test_df = pd.DataFrame(index=range(len(test_list)), columns=['test'])
test_df['test'] = test_list


########## PLOT 1: from Numpy  ##############

to_np = test_df['test'].to_numpy()
print(to_np)

fft_output = np.fft.fft(to_np)
power = np.abs(fft_output)
freq = np.fft.fftfreq(len(to_np))

mask = freq >= 0
freq = freq[mask]
power = power[mask]


plt.plot(freq, power)
plt.show()


########## PLOT 2: from pd.DataFrame  ##############

to_df = test_df['test'].to_frame()
print(to_df)

fft_output = np.fft.fft(to_df)
power = np.abs(fft_output)
freq = np.fft.fftfreq(len(to_df))


mask = freq >= 0
freq = freq[mask]
power = power[mask]


plt.plot(freq, power)
plt.show()
```

### Runtime information:

1.23.5
3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:38:11) 
[Clang 14.0.6 ]

**Comment by seberg** [comment]
Please make sure that the example is complete, ideally without a file, but that file is garbled up so

> From the attached CSV file, [input_data.csv](https://github.com/scipy/scipy/files/11653905/input_data.csv), I import the, say, test_list

is not clear what you mean, how exactly do you load the data?

What does `np.asarray(to_df)` give you?  If it isn't the same as the initial array, that is the reason there you need to investigate.

**Comment by ThemisKoutsellis** [comment]
the `test_list` is a list containing the values of the CSV. See "test" column in the print screen below
![Screenshot 2023-06-05 at 20 08 30](https://github.com/numpy/numpy/assets/88409436/3fae6a95-1ff4-4545-bc64-1225322ea17f)

Therefore,  the `test_df` is a `pd.DataFrame` of the form:
index :      "test"
01/10/14   | 126.209
02/10/14  | 126.677
03/10/14  | 125.083
.....
and so on

**Comment by seberg** [comment]
`12.442.199.999.999.900` is only a valid country in locations where `.` is a thousands delimiter you are asking us to work that out and also the other parameters necessary to load the data.

In any case, I already told you the answer: check what `np.asarray(to_df)` gives you and see that it may be the same numbers, but is not, then see the fft docs with `axis` or pass the series not the dataframe.  NumPy API isn't ideal for dataframes?
Yeah, maybe, but there is nothing we can do about that in NumPy.

**Comment by ThemisKoutsellis** [comment]
> Like most numpy functions expecting an "array_like" input, `np.fft.fft()` will call `np.asarray()` on the passed object. When you give it the `DataFrame` that's the result of `test_df['test'].to_frame()`, the result is a 2-dimensional array of shape `(len(test_df), 1)`, not the `(len(test_df),)` vector that you were expecting. `np.fft.fft()` defaults to performing the FFT over the last axis.

So, in that case ( array of shape (len(test_df), 1)), it is performing against the axis with length=1 and it provides a range of frequencies from 0 to 0.5 ? the same as the transpose case (i.e., `to_np = test_df['test'].to_numpy()`). I really don't get it.

**Comment by ThemisKoutsellis** [comment]
> `12.442.199.999.999.900` is only a valid country in locations where `.` is a thousands delimiter you are asking us to work that out and also the other parameters necessary to load the data.
> 
> In any case, I already told you the answer: check what `np.asarray(to_df)` gives you and see that it may be the same numbers, but is not, then see the fft docs with `axis` or pass the series not the dataframe. NumPy API isn't ideal for dataframes? Yeah, maybe, but there is nothing we can do about that in NumPy.

OK, It was an error of MAC, not a reason to close it. I already gave enough information about what you ask it for (what to_df is). rkern, understood the problem and tried to answer. The problem is that it tries to perform fft to an axis of length=1 given frequencies from 0 to 0.5 (is in 1st plot) which is strange.

**Comment by rkern** [comment]
> So, in that case ( array of shape (len(test_df), 1)), it is performing against the axis with length=1 and it provides a range of frequencies from 0 to 0.5 ?

No, you imposed the frequency range on the first axis rather than the axis the FFT was performed on.
